services/ui_backend_service/data/cache/client/cache_client.py

```diff
@@ -181,7 +183,8 @@ def _call(*args, **kwargs):
             msg, keys, stream_key, disposable_keys =\
                 cls.format_request(*args, **kwargs)
             future = CacheFuture(keys, stream_key, self, cls, self._root)
-            if future.key_paths_ready():
+            invalidate_cache = msg.get('invalidate_cache', False)
+            if future.key_paths_ready() and not invalidate_cache:
```

Comment:
so I'm assuming that the reason the `invalidate_cache` has to be a part of the msg instead of its own return, is so that it can be passed in to the CacheAction `execute()`? Alternatively we could introduce this as part of the CacheAction API, and require the `format_response` to always return a boolean for cache retention, and `execute` to also require it as a parameter.

Current approach works fine as well, just wondering which one makes it more clear. The base CacheAction class gives no hints to the existence of such a feature as is, which would leave new developers unaware of the feature unless they dig deeper under the hood.